python/pyspark/ml/evaluation.py

```diff
@@ -316,17 +335,27 @@ def setParams(self, predictionCol="prediction", labelCol="label",
 
 if __name__ == "__main__":
     import doctest
+    import tempfile
+    import pyspark.ml.evaluation
     from pyspark.context import SparkContext
     from pyspark.sql import SQLContext
-    globs = globals().copy()
+    globs = pyspark.ml.evaluation.__dict__.copy()
```

Comment:
@thunterdb If we did not make this change, we can work well when using ```./bin/pyspark``` to run all tests, but it failed when run with ```./python/run-tests --modules=pyspark-ml```.This issue is weird, I still not figure out the root cause, and we did the same changes when we add other save/load.